utils/components.py

```diff
@@ -298,3 +301,302 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+
+        try:
+            api_response = api_instance.list_namespaced_custom_object(
+                group, version, namespace, plural
+            )
+            print("""
+            ######################### INFO ###########################
+            # Kubeflow Training Operator PyTorchJob CRD is available #
+            ##########################################################
+            """)
+        except ApiException as e:
+            print("""
+            #################################################### ERROR ######################################################################
+            # Kubeflow Training Operator PyTorchJob CRD is unavailable. Ensure your OpenShift AI installation has Training Operator enabled #
+            #################################################################################################################################
+            """)
+            sys.exit(1)
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_oci_model(output_oci_model_uri: str, output_oci_registry_secret: str):
+    import base64
+    import json
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    with open(
+        "/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r"
+    ) as namespace_path:
+        namespace = namespace_path.readline()
+    config.load_incluster_config()
+
+    if output_oci_model_uri is None:
+        print(f"""
+        ############################## INFO ##################################
+        # Parameter output_oci_model_uri not provided. Skipping this step... #
+        ######################################################################
+        """)
+        sys.exit(0)
+    elif not output_oci_model_uri.startswith("oci://"):
+        # If SDG Model Base is not an OCI image, just inform user and quit
+        print(f"""
+        ############################### INFO ####################################
+        # Model {output_oci_model_uri} is not a ModelCar. Skipping this step... #
+        #########################################################################
+        """)
+        sys.exit(0)
+
+    # Extract from sdg_base_model parameter the registry name
+    registry_name = output_oci_model_uri.replace("oci://", "").split("/")[0]
+
+    with client.ApiClient() as api_client:
+        core_api = client.CoreV1Api(api_client)
+        try:
+            secret = core_api.read_namespaced_secret(sdg_oci_docker_secret, namespace)
+            print(f"Reading secret {sdg_oci_docker_secret} data...")
+            if secret.type == "kubernetes.io/dockerconfigjson":
+                # handle authentication if secret provided is kubernetes.io/dockerconfigjson
+                docker_config_json = json.loads(
+                    base64.b64decode(secret.data[".dockerconfigjson"]).decode("utf-8")
+                )
+                if registry_name not in docker_config_json["auths"]:
```

Comment:
This doesn't seem to be permissive enough as wildcards and slashes are accepted:
https://kubernetes.io/docs/concepts/containers/images/#config-json

I found the Kubernetes code for this where `glob` comes from Docker config file and target is the image:
https://github.com/kubernetes/kubernetes/blob/4468565250c940bbf70c2bad07f2aad387454be1/pkg/credentialprovider/keyring.go#L183-L233

This might be too much to reimplement using Python and if it is, I'd be okay with just verifying the secret exists and has a valid Docker config.